### DISCUSSION TITLE
What line actually triggers the fetch in a curl --config file?

### DISCUSSION THREAD

**Discussion by jidanni** [curl/curl#10697] [Q&A] [answered]
It is not clear from the curl --config man page section exactly which
line triggers the fetch. Studying this,
```
# --- Example file ---
# this is a comment
url = "example.com"
output = "curlhere.html"
user-agent = "superagent/1.0"

# and fetch another URL too
url = "example.com/docs/manpage.html"
-O
referer = "http://nowhereatall.example.com/"
# --- End of example file ---
```
my conclusion is that fetches occur when
1) the end of file is reached, or
2) a new url line is reached.

The new url line tells curl to hurry up and process the current url
line, and not keep going along collecting more options. After the
current url is fetched, then it is OK to start working on the next url.

Thus
```
1
2
3
4
url = A
5
6
7
url = B
8
```
lines 1 through 7 will all be options for the fetch of URL A, and B, but 8 will
only affect B.

That's my guess. I sure wish it would be documented there on the man page.

**Comment by bagder** [comment, marked as answer]
The entire file is parsed first and then the transfer(s) start
